src/gtk/toga_gtk/widgets/detailedlist.py

```diff
@@ -1,92 +1,185 @@
+        # refresh button
+        GLib.idle_add(lambda: not self._changed())
+
+    def insert(self, index: int, item: 'Row'):
+        self.store.insert(index, self.row_factory(item))
+        self.list_box.show_all()
+        self._changed()
+
+    def change(self, item: 'Row'):
+        new_item = self.row_factory(item)
+        index = self.index(item)
+        self.store.insert(index, new_item)
+        
+    def remove(self, item: 'Row', index: int):
+        if index is None:
+            index = self.index(item)
+
+        if self._active_row == item._impl:
+            self._active_row = None
+
+        self.store.remove(index)
+        item._impl.destroy()
+        self._changed()
+        
+    def clear(self):
+        self.store.remove_all()
+        self._changed()
 
     def get_selection(self):
-        tree_model, tree_iter = self.selection.get_selected()
-        if tree_iter:
-            row = tree_model.get(tree_iter, 0)[0]
+        item = self.list_box.get_selected_row()
+        if item is None:
+            return item
         else:
-            row = None
+            return item.interface
+
+    def scroll_to_row(self, row: int):
+        item = self.store[row]
+        item.scroll_to_center()
 
-        return row
+    def set_on_refresh(self, handler: callable):
+        if handler is not None:
+            self._on_refresh_handler = handler
+            self.refresh_button.set_on_refresh(self.gtk_on_refresh_clicked)
 
-    def set_on_select(self, handler):
+    def set_on_select(self, handler: callable):
+        self._on_select_handler = handler
+        if handler is not None:
+            self.list_box.set_selection_mode(Gtk.SelectionMode.SINGLE)
+
+    def set_on_delete(self, handler: callable):
+        self._on_delete_handler = handler
+
+    def after_on_refresh(self):
         # No special handling required
         pass
 
-    def set_on_delete(self, handler):
-        pass
+    def index(self, item: 'Row'):
+        for index in range(0, len(self.store)):
+            if item._impl == self.store[index]:
+                return index
+        return None
+
+    def gtk_on_refresh_clicked(self):
+        if self._on_refresh_handler is not None:
+            self._on_refresh_handler(self.interface)
+
+    def gtk_on_row_selected(self, item: Gtk.ListBoxRow):
+        if self._on_select_handler is not None and item is not None:
+            self._on_select_handler(self.interface, item.interface)
+
+        if self._active_row is not None and self._active_row != item:
+            self._active_row.hide_buttons()
+            self._active_row = None
+
+    def gtk_on_delete_clicked(self, item: Gtk.ListBoxRow):
+        if self._on_delete_handler is not None:
+            if self._active_row == item:
+                self._active_row = None
+
+            self.interface.data.remove(item.interface)
+            self._on_delete_handler(self.interface, item.interface)
+
+    def _changed(self):
```

Comment:
As noted above, the naming of this method is a little opaque. There are many things that can change in a widget; it would be helpful if the naming of the method gave some indication of the context of the change where it needs to be invoked (as well as a docstring to clarify further)